backend/TreeOfAKind.Infrastructure/Domain/UserProfiles/UserProfileRepository.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using TreeOfAKind.Domain.UserProfiles;
+using TreeOfAKind.Infrastructure.Database;
+
+namespace TreeOfAKind.Infrastructure.Domain.UserProfiles
+{
+    public class UserProfileRepository : IUserProfileRepository
+    {
+        private readonly TreesContext _dbContext;
+
+        public UserProfileRepository(TreesContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public UserProfile? GetByUsername(string username)
+        {
+            return _dbContext.Users.FirstOrDefault(user => user.Username == username);
```

Comment:
Maybe it would be better to be consistent with asyncs, and accept asyncs in the domain?

Reply:
Todo in future